glib-macros/src/enum_derive.rs

```diff
@@ -65,26 +63,31 @@ fn gen_genum_values(
     )
 }
 
-pub fn impl_genum(input: &syn::DeriveInput) -> TokenStream {
+pub fn impl_enum(input: &syn::DeriveInput) -> TokenStream {
     let name = &input.ident;
 
-    let crate_ident = crate_ident_new();
-
     let enum_variants = match input.data {
         Data::Enum(ref e) => &e.variants,
-        _ => abort_call_site!("GEnum only supports enums"),
+        _ => abort_call_site!("#[derive(glib::Enum)] only supports enums"),
+    };
+
+    let enum_type = match input.attrs.iter().find(|a| a.path.is_ident("enum_type")) {
+        Some(a) => a,
+        _ => abort_call_site!("#[derive(glib::Enum)] requires #[enum_type(\"EnumTypeName\")]"),
     };
 
-    let gtype_name = match parse_type_name(input, "genum") {
-        Ok(v) => v,
+    let gtype_name = match enum_type.parse_args::<LitStr>() {
+        Ok(name) => name,
         Err(e) => abort_call_site!(
-            "{}: derive(GEnum) requires #[genum(type_name = \"EnumTypeName\")]",
+            "{}: #[derive(glib::Enum)] requires #[enum_type(\"EnumTypeName\")]",
```

Comment:
In theory we can also derive all these with a default from the type name btw, but we don't do that intentionally because these need C-style namespacing @pbor

Reply:
yeah, we could pass the namespace instead of the full name, but does not seem to be worth it